src/analysis/processing/qgsalgorithmapproximatemedialaxis.cpp

```diff
@@ -0,0 +1,167 @@
+}
+
+QStringList QgsApproximateMedialAxisAlgorithm::tags() const
+{
+  return QObject::tr( "medial,axis,create,lines,straight,skeleton" ).split( ',' );
+}
+
+QString QgsApproximateMedialAxisAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsApproximateMedialAxisAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsApproximateMedialAxisAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "The Approximate Medial Axis algorithm generates a simplified skeleton of a shape by approximating its medial axis. \n\n"
+                      "The output is a collection of lines that follow the central structure of the shape. The result is a thin, stable set "
+                      "of curves that capture the main topology while ignoring noise.\n\n"
+                      "This algorithm ignores the Z dimensions. If the geometry is 3D, the approximate medial axis will be calculated from "
+                      "its 2D projection." );
+}
+
+QString QgsApproximateMedialAxisAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Returns an approximate medial axis for a polygon layer input based on its straight skeleton." );
+}
+
+QgsApproximateMedialAxisAlgorithm *QgsApproximateMedialAxisAlgorithm::createInstance() const
+{
+  return new QgsApproximateMedialAxisAlgorithm();
+}
+
+QgsFields QgsApproximateMedialAxisAlgorithm::outputFields( const QgsFields &inputFields ) const
+{
+  QgsFields newFields;
+  newFields.append( QgsField( QStringLiteral( "length" ), QMetaType::Type::Double, QString(), 20, 6 ) );
+  return QgsProcessingUtils::combineFields( inputFields, newFields );
+}
+
+QList<int> QgsApproximateMedialAxisAlgorithm::inputLayerTypes() const
+{
+  return QList<int>() << static_cast<int>( Qgis::ProcessingSourceType::VectorPolygon );
+}
+
+QString QgsApproximateMedialAxisAlgorithm::outputName() const
+{
+  return QObject::tr( "Medial axis" );
+}
+
+Qgis::ProcessingSourceType QgsApproximateMedialAxisAlgorithm::outputLayerType() const
+{
+  return Qgis::ProcessingSourceType::VectorLine;
+}
+
+Qgis::WkbType QgsApproximateMedialAxisAlgorithm::outputWkbType( Qgis::WkbType inputWkbType ) const
+{
+  Q_UNUSED( inputWkbType )
+  return Qgis::WkbType::MultiLineString;
+}
+
+bool QgsApproximateMedialAxisAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( parameters )
+  Q_UNUSED( context )
+  Q_UNUSED( feedback )
+
+#ifdef WITH_SFCGAL
+  return true;
+#else
+  throw QgsProcessingException( QObject::tr( "This processing requires a QGIS installation with SFCGAL support enabled. Please use a version of QGIS that includes SFCGAL." ) );
+#endif
+}
+
+QgsFeatureList QgsApproximateMedialAxisAlgorithm::processFeature( const QgsFeature &feature, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( context )
+  Q_UNUSED( feedback )
```

Comment:
It's in use:
```suggestion
```

Reply:
Fixed. It's unused in the without SFCGAL case.